programs/local/LocalServer.cpp

```diff
@@ -227,6 +227,10 @@ void LocalServer::cleanup()
             global_context.reset();
         }
 
+        /// thread status should be destructed before shared context because it relies on process list.
+        thread_status.reset();
+        shared_context.reset();
```

Comment:
Consider just placing `thread_status` and `shared_context` fields in `ClientBase` in reverse order

Reply:
I assume they are already in the right order: `shared_context` in order of destruction goes after `thread_status` in ClientBase. Am I wrong?

Reply:
Hm, yes, so they will be destructed in the right order without this change (but the comment is still useful, we can move it to `ClientBase.h`)